Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1033,7 +1033,7 @@ void HTMLInputElement::setDefaultCheckedState(bool isDefaultChecked)
 void HTMLInputElement::setChecked(bool isChecked, WasSetByJavaScript wasCheckedByJavaScript)
 {
     m_dirtyCheckednessFlag = true;
-    if (checked() == isChecked)
+    if (checked() == isChecked && (!isSwitch() || switchCheckedChangeAnimationProgress() == 1.0f))
         return;
```

Comment:
Are there any implications on event dispatch if we don't early return here during the animation?

Reply:
I don't think so, but I've changed my mind on this particular change. I spend a bunch more time thinking about the overall setup here, but I couldn't really figure out a better alternative. I elaborated on that a bit in the commit message.

Maybe that will come as we add more complexity for iOS and pointer tracking though.